backend/app/api/ws/device_namespace.py

```
@@ -0,0 +1,929 @@
+        if event in self._event_handlers:
+            handler_name = self._event_handlers[event]
+            handler = getattr(self, handler_name, None)
+            if handler:
+                logger.debug(
+                    f"[Device WS] Routing event '{event}' to handler '{handler_name}'"
+                )
+                return await handler(sid, *args)
+
+        return await super().trigger_event(event, sid, *args)
+
+    async def on_connect(self, sid: str, environ: dict, auth: Optional[dict] = None):
```

Ruff (0.14.14)

Unused method argument: `environ`

(ARG002)